lib/logstash/inputs/kafka.rb

```diff
@@ -131,6 +131,8 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
   # The class name of the partition assignment strategy that the client will use to distribute
   # partition ownership amongst consumer instances
   config :partition_assignment_strategy, :validate => :string
+  # ID of the pipeline whose events you want to read from.
+  config :pipeline_id, :validate => :string, :default => "main"
```

Comment:
A potential alternative to to setting this manually in the configuration would be to retrieve the `pipeline_id` from the `execution_context`:

```suggestion
  def pipeline_id
    respond_to?(:execution_context) ? execution_context.pipeline_id : "main"
  end

```